src/impl_methods.rs

```diff
@@ -2250,6 +2230,25 @@ where
         self.fold((), move |(), elt| f(elt))
     }
 
+    /// Call `f` for each element in the array.
+    ///
+    /// Elements are visited in arbitrary order.
+    pub fn for_each_mut<'a, F>(&'a mut self, f: F)
+    where
+        F: FnMut(&'a mut A),
+        A: 'a,
+        S: DataMut,
+    {
+        if self.is_contiguous() {
```

Comment:
To avoid the double check, another version of the `try_into_slice` internal method would be needed (`try_into_memory_order_slice` or similar).

This whole method looks like it could be `self.iter_mut().for_each(f)` except that the iterator is order-preserving 🙁

Reply:
> To avoid the double check, another version of the `try_into_slice` internal method would be needed (`try_into_memory_order_slice` or similar).

Okay, I added a `try_as_slice_memory_order_mut` method to do this.

> This whole method looks like it could be `self.iter_mut().for_each(f)` except that the iterator is order-preserving

Yeah, I've been thinking about the API of the iterators, `.into_shape()`, and other methods where there's a choice of order. One approach would be to introduce a wrapper type for specifying the order of things, similar to your [builder suggestion](https://github.com/rust-ndarray/ndarray/issues/390#issuecomment-761769759), but applied before calling `.iter()`/`.into_shape()`/etc., rather than after. So, it would be used like this: `arr.order(Order::Any).iter()` for a "fast-order" iterator, or like this: `arr.order(Order::Fortran).into_shape(new_shape)` for reshaping with Fortran order.

Reply:
try_as_slice_memory_order_mut  is a nice solution